src/lib/pickles/step_main.ml

```diff
@@ -44,7 +44,7 @@ let step_main :
            Types.Pairing_based.Statement.t
         -> unit )
        Staged.t =
- fun (module Req) (module Max_branching) ~self_branches ~local_signature
+ fun (module Req) max_branching ~self_branches ~local_signature
```

Comment:
Why is this change necessary?

Reply:
Because of the reasons outlined in this OCaml issue: https://github.com/ocaml/ocaml/issues/10768
Basically the typechecking is a bit more strict in situations where first-class modules are unpacked too early and indeed you'll find that OCaml 4.14 complains about invalid type equalities without these changes.

Every change I made to `pickles` is because of this. I couldn't find a solution more elegant than this + the `(module struct include Max_branching end)` when we want to subtype the interface (from `Nat.Add.Intf` to `Nat.Intf`, most of the time)